docs/openshift-ai/data-science-project/testing-model-in-the-rhoai.md

```diff
@@ -0,0 +1,143 @@
+
+    ![Jupyter Hub Control Panel Menu](images/juyter-hub-control-panel-menu.png)
+
+- In your project in JupyterLab, open the notebook `03_remote_inference.ipynb` and
+follow the instructions to see how the model can be queried.
+
+- Update the `grpc_url` as [noted before](model-serving-in-the-rhoai.md#deploy-the-model)
+for the **the grpc URL value** from the deployed model on the NERC RHOAI Model server.
+
+    ![Change grpc URL Value](images/change-grpc-url-value.png)
+
+- Once you've completed the notebook's instructions, the object detection model
+can isolate and recognize T-shirts, bottles, and hats in pictures, as shown below:
+
+    ![Model Test to Detect Objects In An Image](images/model-test-object-detection.png)
+
+## Building and deploying an intelligent application
+
+The application we are going to deploy is a simple example of how you can add an
+intelligent feature powered by AI/ML to an application. It is a webapp that you
+can use on your phone to discover coupons on various items you can see in a store,
+in an augmented reality way.
+
+### Architecture
+
+The different components of this intelligent application are:
+
+• **The Frontend**: a React application, typically running on the browser of your
+phone,
+
+• **The Backend**: a NodeJS server, serving the application and relaying API calls,
+
+• **The Pre-Post Processing Service**: a Python FastAPI service, doing the image
+pre-processing, calling the model server API, and doing the post-processing before
+sending the results back.
+
+• **The Model Server**: the RHOAI component serving the model as an API to do
+the inference.
+
+### Application Workflow Steps
+
+1. Pass the image to the pre-post processing service
+
+2. Pre-process the image and call the model server
+
+3. Send back the inference result
+
+4. Post-process the inference and send back the result
+
+5. Pass the result to the frontend for display
+
+### Deploy the application
+
+The deployment of the application is really easy, as we already created for you
+the necessary YAML files. They are included in the Git project we used for this
+example project. You can find them in the **deployment** folder inside you JupyterLab
+environment, or directly [here](https://github.com/nerc-project/nerc_rhoai_mlops/tree/main/deployment).
+
+To deploy the Pre-Post Processing Service service and the Application:
+
+- From your [NERC's OpenShift Web Console](https://console.apps.shift.nerc.mghpcc.org/),
+navigate to your project corresponding to the *NERC RHOAI Data Science Project*,
+and select the "Import YAML" button, the "plus" sign on the top right:
+
+    ![YAML Add Icon](images/yaml-upload-plus-icon.png)
+
+- Copy/Paste the content of the file `pre_post_processor_deployment.yaml` inside
+the opened YAML editor. If you have named your model **coolstore** as instructed,
+you're good to go. If not, modify the value on **line # 35** with the name you set.
```

Comment:
will it always be line 35? can we point to something else unique on the row?

Reply:
Yes. Now I added link to the versioned github url with line # that will be fixed and not changed, even we change the code later time.